packages/modules/src/dataSelection/DataSourceRow.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react'
+import { Utf8, Utf8Vector } from 'apache-arrow'
+import { RowLike } from 'apache-arrow/type'
+
+import TableRow from '@material-ui/core/TableRow'
+import TableCell from '@material-ui/core/TableCell'
+import Tooltip from '@material-ui/core/Tooltip'
+import Checkbox from '@material-ui/core/Checkbox'
+
+import { useDataRepositoryItemValue, DataRepositoryItemsTable } from '@dharpa-vre/client-core'
+import { LoadingIndicator } from '@dharpa-vre/client-ui'
+
+import useStyles from './DataSourceRow.styles'
+
+import { TableView } from '../components/TableView'
+
+type DataSourceRowProps = {
+  repositoryItemBatch: DataRepositoryItemsTable
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  repositoryItem: RowLike<any>
+  rowIndex: number
+  selectedItemIds: string[]
+  setSelectedItemIds: (value: string[]) => Promise<void>
+}
+
+const DataSourceRow = ({
+  repositoryItemBatch,
+  repositoryItem,
+  rowIndex,
+  selectedItemIds,
+  setSelectedItemIds
+}: DataSourceRowProps): JSX.Element => {
+  const classes = useStyles()
+
+  const dataSourceId = String(repositoryItemBatch?.getColumn('id').toArray()[rowIndex])
+  const [dataSourceContentTable, dataSourceContentMetadata] = useDataRepositoryItemValue(
```

Comment:
Just for information, this will be moved up the abstraction levels soon because the preview item may be of other types.